File: backend/pkg/model/message.go
package model

type Message struct {
	ID           int    `json:"id"`
	SenderId     int    `json:"sender_id"`
	RecipientId  int    `json:"recipient_id"`
	IsOwned		 bool 	`json:"isOwned"`
	GroupId      int    `json:"group_id"`
	IsSeen       int    `json:"is_seen"`
	Text         string `json:"text"`
	User 	*User	`json:"user"`
	CreationDate string `json:"created_at"`
}

File: frontend/types/message.ts
import { User } from "./user";

export interface Message {
  id: number;
  sender_id: number;
  recipient_id?: number;
  group_id?: number;
  text: string;
  created_at: string;
  isOwned: boolean;
  user: User;
}

export interface AddMessageEvent {
  type: string;
  message: Message;
  isOwned: boolean;
}
